treeherder/config/settings.py

```diff
@@ -88,6 +88,15 @@
     # 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 ]
 
+if DEBUG:
+    # insert debug toolbar middleware just after the gzip
+    # middleware
+    gzip_index = MIDDLEWARE_CLASSES.index(
+        'django.middleware.gzip.GZipMiddleware')
+    MIDDLEWARE_CLASSES.insert(
+        gzip_index + 1,
+        'debug_toolbar.middleware.DebugToolbarMiddleware')
```

Comment:
Though using filter is 9x faster than the bedrock approach:
http://stackoverflow.com/a/3845453

Also at the moment, if we set `DEBUG` anywhere other than locally, it will cause failures. I don't think it always holds that is_vagrant == DEBUG (eg we might in a rare case want to enable it on stage).

An alternative `settings.py` addition might be something like...
```python
ENABLE_DEBUG_TOOLBAR = env.bool("ENABLE_DEBUG_TOOLBAR", False)

MIDDLEWARE_CLASSES = filter(None, [
    ...
    'django.middleware.gzip.GZipMiddleware',
    'debug_toolbar.middleware.DebugToolbarMiddleware' if ENABLE_DEBUG_TOOLBAR else None,
    ...
    # 'django.middleware.clickjacking.XFrameOptionsMiddleware',
])

if ENABLE_DEBUG_TOOLBAR:
    INSTALLED_APPS.append('debug_toolbar')
```
...where `ENABLE_DEBUG_TOOLBAR` is defined in `vagrant.pp` along with the other local environment variables.

Anyway, happy to leave it to you to decide :-)